libautoscoper/src/Tracker.cpp

```diff
@@ -442,7 +448,26 @@ void Tracker::optimize(int frame, int dFrame, int repeats, int opt_method, unsig
 
       clock_t cpu_begin = clock();
 
-      pso(positions, velocities, pBests, gBest, MAX_EPOCHS, MAX_STALL);
+      // check for the environment variable Autoscoper_PSO_MAIN_OUTPUT_DIR
+      if (char* dirname = getenv("Autoscoper_PSO_MAIN_OUTPUT_DIR")) {
+        string output_dir(dirname);
+        cerr << "Autoscoper_PSO_MAIN_OUTPUT_DIR is set to " << output_dir << endl;
+        output_dir += (OS_SEP + string("Volume_") + to_string(trial_.current_volume));
+        output_dir += (OS_SEP + string("frame_") + to_string(frame));
+        filesystem::path p(output_dir);
+        error_code ec;
+        if (!filesystem::is_directory(p) || !filesystem::exists(p)) {
+            filesystem::create_directories(p,ec);
+            if (ec) {
+				cerr << "Error creating directory " << output_dir << ": " << ec.message() << endl;
+                exit(1);
+			}
+        }
+        output_dir += OS_SEP;
```

Comment:
I suggest to move this directly into the `pso()` function

Reply:
In order to format the directories with the relevant volume and frame, this would need to be in the tracker file since the `pso()` function does not have access to this information.